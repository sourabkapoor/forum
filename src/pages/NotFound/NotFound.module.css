.not-found {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    row-gap: 10px;
    color: var(--secondary-color);
    height: 100%;
    padding: 6rem 2rem;
    width: 100%;
}

.not-found__image {
    border-radius: 9999px;
}

.not-found__title {

}