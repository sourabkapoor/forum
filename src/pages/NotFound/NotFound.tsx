import { useEffect } from "react";
import CSSModules from "react-css-modules";
import { useNavigate } from "react-router-dom";
import styles from "./NotFound.module.css";
const NotFound: React.FC = () => {
  const navigate = useNavigate();

  useEffect(() => {
    setTimeout(() => {
      navigate("/");
    }, 5000);
  }, [navigate]);

  return (
    <div styleName="not-found">
      {/* <img
        src="https://www.redditstatic.com/reddit404d.png"
        styleName="not-found__image"
        alt="redditor who is lost"
      /> */}
      <h1 styleName="not-found__title">404: page not found</h1>
      <p styleName="not-found__description">
        the page you requested does not exist...redirecting
      </p>
    </div>
  );
};

export default CSSModules(NotFound, styles, {
  allowMultiple: true,
  handleNotFoundStyleName: "ignore",
});
