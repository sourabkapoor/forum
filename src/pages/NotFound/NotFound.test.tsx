import {
  queryByPlaceholderText,
  render,
  screen,
  waitFor,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import each from "jest-each";
import NotFound from "./NotFound";
import { BrowserRouter } from "react-router-dom";

describe("NotFound", () => {
  test("snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("user is navigated after a certain time", async () => {

    const mockNavigate = jest.fn()
    jest.mock("react-router-dom", () => ({
      ...jest.requireActual("react-router-dom"),
      useNavigate: () => mockNavigate
    }))

    render(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>
    );

    await waitFor(() => {
      expect(mockNavigate).toHaveBeenCalledTimes(1);
    });
  });
});
