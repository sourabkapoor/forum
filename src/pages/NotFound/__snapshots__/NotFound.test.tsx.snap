// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound snapshot 1`] = `
<DocumentFragment>
  <div
    class="not-found"
  >
    <img
      alt="redditor who is lost"
      class="not-found__image"
      src="https://www.redditstatic.com/reddit404d.png"
    />
    <h1
      class="not-found__title"
    >
      404: page not found
    </h1>
    <p
      class="not-found__description"
    >
      the page you requested does not exist...redirecting
    </p>
  </div>
</DocumentFragment>
`;
